### ISSUE TITLE
Final edits (typos, grammar, etc.) to be made to spec prior to final publication

### ISSUE THREAD

**Issue by wmhopkins** [javaee/security-spec#62]
From Rudy's review:
- On page 18 there are a few small issues with formatting (unbalanced underscores for italic) in the first 3 paragraph/bullet points.

From Arjan:
- "This chapter overview information and terminology related to this specification, and also includes a general requirements not specified elsewhere in this document." -> should say something like, "This chapter contains an overview of information ..."  and "includes general requirements ..."
- "validateRequest() will be invoked before the doFilter() method of any servlet filter or the service() method of any servlet in the application for requests to constrained as well as to unconstrained resources, and, in addition, in response to application code calling the authenticate() method on the HttpServletRequest." -> For completeness this should maybe include SecurityContext as well.
- On page 21, we forgot to update this status to the renamed version; "SEND_CONTINUE":
```
if (status.equals(IN_PROGRESS)) {
    facesContext.responseComplete();
}
```
From Werner:
- Nevertheless, I believe JDBC 4 should be mentioned and a reference added in 3.5 as the `@DatabaseIdentityStoreDefinition` is part of chapter 3.

Credits:
- Credit Alex for his work as spec lead.
- Credit Ron for his work laying down the foundation with earlier JSRs. 

Bugs:
- Documentation for search filters needs to document %s syntax, and maybe the requirement for expression to be enclosed in parens.
- Doc for `LdapIdentityStore` says the value of `callerNameAttribute` is returned as the principal name, but that's not true -- it returns the username from the `UsernamePasswordCredential`.
- Default behaviors (user/group search filters) on LDAP id store not documented
- `SecurityContext` javadoc says it's available in all containers, but spec says web and EJB only, others optional.
- `getCallerPrincipal()` description in spec doesn't define "same as servlet/ejb" semantics, like `isCallerInRole()` does.
- Javadoc should not `{@link javax.resource.spi.AuthenticationMechanism}`, as that annotation is at best conceptually similar -- `HttpAuthenticationMechanism` is not an `AuthenticationMechanism`, and the two types have no actual relationship to each other.
- default impl for HttpAuthenticationMechanism.secureResponse() should return SEND_SUCCESS, not SUCCESS.

**Comment by arjantijms** [comment]
>Neither spec nor javadoc documents that the identity store validate method throws an exception on error, rather than return a status.

Not 100% sure this is indeed a bug.

>Javadoc should not {@link javax.resource.spi.AuthenticationMechanism}, as that annotation is at best conceptually similar -- HttpAuthenticationMechanism is not an AuthenticationMechanism, and the two types have no actual relationship to each other.

This one should most likely be removed indeed.

**Comment by wmhopkins** [comment]
> Neither spec nor javadoc documents that the identity store validate method throws an exception on error, rather than return a status.
Not 100% sure this is indeed a bug.

Agreed, although if validate() can through an unchecked error, it would be best for the javadoc to say so. I'll remove it.

The real issue is that we haven't specified the expected behavior in the face of errors; I opened a separate issue to deal with this ( #63 ).

**Comment by wmhopkins** [comment]
Fixed javadoc issue with `{@link javax.resource.spi.AuthenticationMechanism}`.

**Comment by wmhopkins** [comment]
Fixed typos and acknowledgements with PR #64; remaining issues couldn't be fixed since the spec is final, but created #65 to track them for a future spec update.
